--- src/mpid/ch3/channels/psm/src/psm_vbuf.c.orig	2014-04-28 13:46:18.077708179 -0400
+++ src/mpid/ch3/channels/psm/src/psm_vbuf.c	2014-04-28 13:49:01.056609252 -0400
@@ -78,8 +78,8 @@ void psm_deallocate_vbuf()
     while (r)
     {
         next = r->next;
-        free(r->malloc_start);
-        free(r->malloc_buf_start);
+        (free)(r->malloc_start);
+        (free)(r->malloc_buf_start);
         MPIU_Free(r);
         r = next;
     }
--- src/mpid/ch3/channels/psm/src/psm_entry.c.orig	2014-04-28 13:42:46.053733555 -0400
+++ src/mpid/ch3/channels/psm/src/psm_entry.c	2014-04-28 13:43:46.522298521 -0400
@@ -294,9 +294,9 @@ static void psm_preinit(int pg_size)
         for(i = 0; i < n; i++) {
             if(0 == strcmp(scratch, fls[i]->d_name))
                 id = i;
-            free(fls[i]);
+            (free)(fls[i]);
         }   
-        free(fls);
+        (free)(fls);
 
         PMI_Barrier();
         DBG("localid %d localranks %d\n", id, n);
